skyreader/plot/plot.py

```diff
@@ -816,39 +821,25 @@ def bounding_box(ra, dec, theta, phi):
                     color=cont_col,
                     linestyle=cont_sty
                 )
-
         plt.legend(fontsize=6, loc="lower left")
-
         # Dump the whole contour
         path = unique_id + ".contour.pkl"
         print("Saving contour to", path)
         with open(path, "wb") as f:
             pickle.dump(saving_contours, f)
 
-        # logic for multiordermaps -> for future developements
-
-        # save multiorder version of the map
-        # multiorder_map = mhealpy.HealpixMap(
-        #     grid_value, uniq_array
-        # )
-        # multiorder_map.write_map(
-        #     f"{unique_id}.skymap_nside_{mmap_nside}.multiorder.fits.gz",
-        #     column_names=["PROBABILITY"],
-        #     extra_header=fits_header,
-        #     overwrite=True,
-        # )
-
         if llh_map:
             column_names = ['2DLLH']
         else:
             # avoid excessively heavy data format for the flattened map
-            equatorial_map[
-                equatorial_map < 1e-16
-            ] = np.mean(
+            min_prob = np.nanmin(equatorial_map)
+            equatorial_map[np.isnan(equatorial_map)] = np.min(
+                [min_prob, 1e-16]
+            )
+            equatorial_map[equatorial_map < 1e-16] = np.mean(
```

Comment:
I see why having many different float values can bloat the file size, and that is why we clipped it to the mean. But why is it an issue with the nans? Also, with the multi order coverage maps do we still need to clip it? 

I would think it's cleaner to 
- preserve the nans in the multi-resolution/flattened maps
- remove the np.mean clipping, which is an artifact of the applying the convolution, and let the size reduction be handled by the multi-resolution maps.

Reply:
I made some modifications so that:
- Only before the convolution, I use `np.nan_to_num` (as suggested by you) to transform nans in zeros.
- In all the rest of the code, nans are now kept.
- The clipping is applied only to the flattened map, the multi-resolution map has no clipping (flattened map saves `equatorial_map`, multi-resolution map saves `grid_value`)